debezium-connector-binlog/src/main/java/io/debezium/connector/binlog/BinlogStreamingChangeEventSource.java

```diff
@@ -758,15 +760,16 @@ protected void handleQueryEvent(P partition, O offsetContext, Event event) throw
                         && schemaChangeEvent.getType().equals(SchemaChangeEvent.SchemaChangeEventType.TRUNCATE)) {
                     eventDispatcher.dispatchDataChangeEvent(partition, tableId,
                             new BinlogChangeRecordEmitter<>(partition, offsetContext, clock, Envelope.Operation.TRUNCATE, null, null, connectorConfig));
+                } else {
+                    eventDispatcher.dispatchSchemaChangeEvent(partition, offsetContext, tableId, (receiver) -> {
+                        try {
+                            receiver.schemaChangeEvent(schemaChangeEvent);
+                        } catch (Exception e) {
+                            throw new DebeziumException(e);
+                        }
+                    });
```

Comment:
This won't be sufficient to exclude the `TRUNCATE` event in all cases.
If the skipped.operations have `TRUNCATE` included, this will produce it into the schema history topic.

Better to explicitly have an if condition to exclude the `TRUNCATE` entry.